src/lib/utils/solidity/remapImports.ts

```diff
@@ -0,0 +1,144 @@
+  }
+
+  // http import
+  if (isHttp(importPath)) {
+    return new File({ externalUrl: importPath, type: Types.http });
+  }
+
+  // imported from node_modules, ie import "@aragon/os/contracts/acl/ACL.sol"
+  if (isNodeModule(importPath)) {
+    from = path.join("node_modules", importPath);
+    to = path.join(".embark", from);
+    fs.copySync(from, to);
+    return new File({ path: to, type: Types.dappFile });
+  }
+
+  // started with node_modules then further imports local paths in it's own repo/directory
+  if (isEmbarkNodeModule(file.path)) {
+    from = path.join(path.dirname(file.path.replace(".embark", ".")), importPath);
+    to = path.join(path.dirname(file.path), importPath);
+    fs.copySync(from, to);
+    return new File({ path: to, type: Types.dappFile });
+  }
+
+  // local import, ie import "../path/to/contract" or "./path/to/contract"
+  from = path.join(path.dirname(file.path.replace(".embark", ".")), importPath);
+  to = path.join(".embark", from);
+
+  fs.copySync(from, to);
+  return new File({ path: to, type: Types.dappFile });
+};
+
+const rescursivelyFindRemapImports = async (file: File) => {
+  let remapImports: RemapImport[] = [];
+  const content = await file.content;
+  const imports = getImports(content);
+
+  // if no imports, break recursion
+  if (!imports.length) {
+    return [];
+  }
+
+  for (const importPath of imports) {
+    const newFile = buildNewFile(file, importPath);
+    file.importRemappings.push({prefix: importPath, target: newFile.path});
+    remapImports.push({path: file.path, searchValue: importPath, replaceValue: newFile.path});
+    remapImports = remapImports.concat(
+      await rescursivelyFindRemapImports(newFile),
+    );
+  }
+
+  return remapImports;
+};
+
+const isEmbarkNodeModule = (input: string) => {
+  return input.startsWith(".embark/node_modules");
+};
+
+const isNodeModule = (input: string) => {
+  return !input.startsWith("..") && fs.existsSync(path.join("./node_modules/", input));
+};
+
+const isHttp = (input: string) => {
+  return input.startsWith("https://") || input.startsWith("http://");
+};
+
+const replaceImports = (remapImports: RemapImport[]) => {
+  const byPath: {[path: string]: [{searchValue: string, replaceValue: string}]} = groupBy(remapImports, "path");
+  Object.keys(byPath).forEach((p) => {
+    let source = fs.readFileSync(p, "utf-8");
+    byPath[p].forEach(({searchValue, replaceValue}) => {
+      source = source.replace(`import "${searchValue}";`, `import "${replaceValue}";`);
+    });
+    fs.writeFileSync(p, source);
+  });
+};
+
+export const prepareForCompilation = async (file: File, isCoverage = false) => {
+  await prepareInitialFile(file);
+  const remapImports = await rescursivelyFindRemapImports(file);
+  replaceImports(remapImports);
+
+  const content = await file.content;
+  if (!isCoverage) {
+    return content;
+  }
+
+  removePureView(path.join(".embark"));
+  return replacePureView(content);
```

Comment:
Why do we have to do that? Isn't it only needed for coverage?

Reply:
Look just above:
if (!coverage)

Also that needs to be done for the contract that we return modified